packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -430,6 +430,8 @@ abstract class ImplicitlyAnimatedWidgetState<T extends ImplicitlyAnimatedWidget>
         tween ??= constructor(targetValue);
         if (_shouldAnimateTween(tween, targetValue))
           shouldStartAnimation = true;
+        else
+          tween.end ??= tween.begin;
```

Comment:
I don't think this is correct. For some tweens (e.g. ColorTween) null has a special meaning. And as documented in https://master-api.flutter.dev/flutter/animation/Tween-class.html under "Nullability" that's totally ok.

Reply:
I understand that.

1. I haven't removed the ability for the value to be `null`. I'm just making sure that when the tween should not animate, begin and end should be the same, as there is no animation. This adds to a state of being unanimated.
2. In the current implementation of `ImplicitlyAnimatedWidget`, it is not possible to animate from non-null color to null anyway, as when the color (aka `targetValue`) becomes null, the tween is set to null, ignoring it. As a matter of fact, this is something that needs to be fixed too.
3. Lastly, `ImplicitlyAnimatedWidget` should be agnostic of what should happen when `targetValue` becomes null. Current extensions of `ImplicitlyAnimatedWidget` are trying to expect different things from it. Example: `AnimatedContainer` has a nullable `alignment` argument which is supposed to be ignored when it is null, which makes its tween null. Whereas `AnimatedColor` (fictional widget working over `ColorTween`) has a special meaning when its `color` is null, meaning its tween should remain non-null and have `end` set to null.

Possible solution:
1. The `tween`s' nullability should be handled by `ImplicitlyAnimatedWidget`'s extension instead of itself.
2. The check for whether a tween should animate or not, should depend only on its current `end`, even it it's null.
    ```dart
    bool _shouldAnimateTween(Tween<dynamic> tween, dynamic targetValue) {
        return targetValue != tween.end;
      }
    ```
3. From the perspective of `ImplicitlyAnimatedWidget`, `targetValue` being null should be considered as one of the valid tween states instead of using it to null the tween.